ludwig/utils/trainer_utils.py

```diff
@@ -408,3 +410,20 @@ def get_rendered_batch_size_grad_accum(config: "BaseTrainerConfig", num_workers:
                 gradient_accumulation_steps = 1
 
     return batch_size, gradient_accumulation_steps
+
+
+def freeze_layers_regex(config: "BaseTrainerConfig", model: ECD) -> None:
+    """Freezes layers based on provided regular expression."""
+    try:
+        pattern = re.compile(config.layers_to_freeze_regex)
+    except re.error:
+        logger.error(f"Invalid regex input: {config.layers_to_freeze_regex}")
+        exit()
```

Comment:
Instead of `exit()`, let's raise a RuntimeError() with the same message. 

In fact, here's a thought I have: We can move this check to earlier in the code path, that is, at config validation time. Specifically, you can create a `__post_init__()` hook for `ECDTrainerConfig` and `FineTuneTrainerConfig` that tries to do re.compile() and if it fails, throws a ConfigValidationError with the error message. That way, we don't have to wait for all of preprocessing etc to be done before catching this error. 

Here's an example explaining the same idea in a different part of the Ludwig codepath: https://github.com/ludwig-ai/ludwig/blob/master/ludwig/schema/llms/peft.py#L443